### PR TITLE
always publish datadog.agent.checks.runs metrics, not just in agent t…

### PR DESCRIPTION
**Pull request by djmitche**
### What does this PR do?

Adds `datadog.agent.checks.runs` to the built-in agent telemetry sent to the backend

### Motivation

Should help with debugging check-related issues, to see if the checks are running and if they are failing.

### Additional Notes

This removes the same metrics from "agent telemetry", to avoid duplication.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/checks-runs-publish-3accc6aac8c5cf93.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+features:
+  - |
+    The agent now generates a metric, ``datadog.agent.checks.runs``, giving the
```
Comment (apigirl) [inline releasenotes/notes/checks-runs-publish-3accc6aac8c5cf93.yaml:3]:
```suggestion
    The Agent now generates a metric, ``datadog.agent.checks.runs``, giving the
```

**Review by apigirl — COMMENTED** [review]
just one small nit for the release note editorial review

**Review by vickenty — APPROVED** [review]

**Review by truthbk — APPROVED** [review]

**Review by olivielpeau — CHANGES_REQUESTED** [review]
We need to further discuss the cardinality of this new metric before merging this

**Review by apigirl — APPROVED** [review]
release note looks good

**Pull request closed without merging**